website/blog/documenting-an-express-api-with-optic.md

```diff
@@ -0,0 +1,396 @@
+      try {
+         const comment = {}
+         comment.title        = req.body.title;
+         comment.body         = req.body.body;
+         comment.articleImage = req.body.article_image
+ 
+         const updatedArticle = await ArticleService.updateArticle(comment);
+ 
+         if(updatedArticle.modifiedCount === 0){
+            throw new Error("Unable to update article, error occord");
+         }
+ 
+         res.json(updatedArticle);
+ 
+      } catch (error) {
+         res.status(500).json({error: error});
+      }
+   }
+ 
+   static async apiDeleteArticle(req, res, next){
+         try {
+            const articleId = req.params.id;
+            const deleteResponse =  await ArticleService.deleteArticle(articleId)
+            res.json(deleteResponse);
+         } catch (error) {
+            res.status(500).json({error: error})
+         }
+ }
+ 
+}
+```
+
+Your routes are defined on the `articles.routes.js` file, while the controllers are defined on the `article.controller.js` file. The only file yet to be updated is your articleService.js` file. Go ahead and update it with the following code:
+
+```js title='articles.routes.js'
+const Article = require("./Article");
+ 
+module.exports = class ArticleService{
+    static async getAllArticles(){
+        try {
+            const allArticles = await  Article.find();
+            return allArticles;
+        } catch (error) {
+            console.log(`Could not fetch articles ${error}`)
+        }
+    }
+ 
+    static async createArticle(data){
+        try {
+ 
+            const newArticle = {
+                title: data.title,
+                body: data.body,
+                article_image: data.article_image
+            }
+           const response = await new Article(newArticle).save();
+           return response;
+        } catch (error) {
+            console.log(error);
+        } 
+ 
+    }
+    static async getArticlebyId(articleId){
+        try {
+            const singleArticleResponse =  await Article.findById({_id: articleId});
+            return singleArticleResponse;
+        } catch (error) {
+            console.log(`Article not found. ${error}`)
+        }
+    }
+ 
+    static async updateArticle(title, body, articleImage){
+            try {
+                const updateResponse =  await Article.updateOne(
+                    {title, body, articleImage}, 
+                    {$set: {date: new Date.now()}});
+ 
+                    return updateResponse;
+            } catch (error) {
+                console.log(`Could not update Article ${error}` );
+ 
+        }
+    }
+ 
+    static async deleteArticle(articleId){
+        try {
+            const deletedResponse = await Article.findOneAndDelete(articleId);
+            return deletedResponse;
+        } catch (error) {
+            console.log(`Could  ot delete article ${error}`);
+        }
+ 
+    }
+}
+```
+
+Now let's pause and restart the server. Proceed to `localhost:8000/API/v1/articles`.
```

Comment:
It looks like you ran the steps here to generate the images @LouManglass so if the capitalization was important this might need to be fixed

Reply:
ooh, nice, yep. I'll get that resolved.